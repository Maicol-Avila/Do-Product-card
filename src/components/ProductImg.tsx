import React, { useContext } from 'react';
import noMoule from '../assets/no-image.jpg';
import style from '../styles/styles.module.css';
import { ProductContext } from './ProductCard';
import { PropsImage } from '../interface/interfaces';

export const ProductImg = ({ img, className, styles }: PropsImage) => {
  const { product } = useContext(ProductContext);

  let imgShow: string;

  if (img) {
    imgShow = img;
  } else if (product.img) {
    imgShow = product.img;
  } else {
    imgShow = noMoule;
  }

  return (
    <img
      className={`${style.productImg} ${className}`}
      style={styles}
      src={imgShow}
      alt="Coffe mug"
    />
  );
};

export default ProductImg;
