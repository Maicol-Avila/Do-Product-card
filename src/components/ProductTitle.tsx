import React , { useContext } from 'react';
import { PropsTitle } from '../interface/interfaces';


import style from '../styles/styles.module.css';
import { ProductContext } from './ProductCard';

export const ProductTitle = ({ title, className }: PropsTitle) => {
  const { product } = useContext(ProductContext);

  return (
    <span 
    className={` ${style.productDescription} ${className}`}
    >
      {title ? title : product.title}
    </span>
  );
};

export default ProductTitle;
