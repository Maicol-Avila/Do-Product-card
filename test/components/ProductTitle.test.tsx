import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {

  test('Debe mopstar el componente del titulo perzonalizado', () => {
    const wrapper = renderer.create(
    
    <ProductTitle title="Custom product" className='cusmtom-class' />
    
    );
    expect(wrapper.toJSON()).toMatchSnapshot()
  });
  
    
    test('Debe mostrarse el componente con el nombre', () => {
      const wrapper = renderer.create(
        <ProductCard  product={product1}> 
          {
            () => (
              <ProductTitle/>
            )
          }
        </ProductCard>
        
      );
      expect(wrapper.toJSON()).toMatchSnapshot()
    })

});
