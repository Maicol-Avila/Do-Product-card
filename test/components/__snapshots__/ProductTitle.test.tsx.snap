// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe mopstar el componente del titulo perzonalizado 1`] = `
<span
  className=" productDescription cusmtom-class"
>
  Custom product
</span>
`;

exports[`ProductTitle Debe mostrarse el componente con el nombre 1`] = `
<div
  className=" productCard undefined "
  id="1"
>
  <span
    className=" productDescription undefined"
  >
    Este es el titulo del producto 1
  </span>
</div>
`;
